Add ICampaignService and CampaignService (waiting for implementation)

mbily opened the pull request.

Commit:
Add ICampaignService and CampaignService (waiting for implementation)

mozdren commented on ServiceBus/ServiceBus/CampaignService.cs:
```diff
@@ -0,0 +1,47 @@
+        public SharedLibs.DataContracts.Campaign GetCampaign(Guid guid)
+        {
+            //before continuing is necessary to add Campaign to ServiceBusDatabaseEntities
+            throw new NotImplementedException();
```
I would preffer having something like:

return new Result.Fatal("Not Implemented");

This should not throw an exception, but rather give the client information that the service method is not implemented yet. Please do that for all methods. I will approve after

mozdren commented on ServiceBus/ServiceBus/CampaignService.cs:
```diff
@@ -0,0 +1,47 @@
+        public SharedLibs.DataContracts.Campaigns GetAllCampaign()
+        {
+            //before continuing is necessary to add Campaign to ServiceBusDatabaseEntities
+            throw new NotImplementedException();
```
return new Result.Fatal...

mozdren commented on ServiceBus/ServiceBus/CampaignService.cs:
```diff
@@ -0,0 +1,47 @@
+        public SharedLibs.DataContracts.Result AddCampaign(Campaign campaign)
+        {
+            //before continuing is necessary to add Campaign to ServiceBusDatabaseEntities
+            throw new NotImplementedException();
```
return new Result.Fatal...

mozdren commented on ServiceBus/ServiceBus/CampaignService.cs:
```diff
@@ -0,0 +1,47 @@
+        public SharedLibs.DataContracts.Result AddCampaign(Guid guid, string name, double discount)
+        {
+            //before continuing is necessary to add Campaign to ServiceBusDatabaseEntities
+            throw new NotImplementedException();
```
return new Result.Fatal...

mozdren commented on ServiceBus/ServiceBus/CampaignService.cs:
```diff
@@ -0,0 +1,47 @@
+        public Result EditCampaign(Guid guid, string name, double discount)
+        {
+            //before continuing is necessary to add Campaign to ServiceBusDatabaseEntities
+            throw new NotImplementedException();
```
return new Result.Fatal...

mozdren commented on ServiceBus/ServiceBus/CampaignService.cs:
```diff
@@ -0,0 +1,47 @@
+        public Result DeleteCampaign(Guid guid)
+        {
+            //before continuing is necessary to add Campaign to ServiceBusDatabaseEntities
+            throw new NotImplementedException();
```
return new Result.Fatal...

mozdren commented on ServiceBus/ServiceBus/ICampaignService.cs:
```diff
@@ -0,0 +1,65 @@
+        /// </summary>
+        /// <returns>collection of all campaigns</returns>
+        [OperationContract]
+        SharedLibs.DataContracts.Campaigns GetAllCampaign();
```
rename to GetAllCampaigns -> the 's' at the end

mozdren reviewed (changes requested):
Please, update the code according to comments I did and I can approve. Thanks.

mozdren commented:
Also, please create a Mock implementation for the campaign service (it should use the same interface but should have implementation which returns some fake data) - see the ProductService in the Mock forlder.

Commit:
Fixed bugs according to comments in the request #5
CampaignService changed from class (CampaignService.cs) to WCF Service (CampaignService.svc.cs)

Commit:
Mock service for CampaignService added

mozdren reviewed (approved):
Good job!

mozdren commented:
It has to be merged manually

mozdren committed:
Merge - master to branch
